docs/Error-Handling.md

```diff
@@ -0,0 +1,37 @@
+<h1 align="center">Fastify</h1>
+
+## Error classification in Promises & Async/Await
+
+Fastify classified errors in two different groups.
+
+- `System` any instance of type `EvalError`, `RangeError`, `ReferenceError`, `SyntaxError`, `TypeError`, `URIError` or `AssertionError`
+- `Business` any instance of type `Error`
+
+The main difference between those errors are that `System` errors are handled by fastify and result in an [`unhandled rejection`](#unhandled-rejections).
+
+### Background
+
+With the use of `promises` and `async/await` we are able to catch **any** synchronous error and respond it back to the callee doesn't matter whether it's correct behavior or a syntax error. This is an anti-pattern because any unpredictable application state should exit your server so that it can come back in a clear state. This is useful for debugging and can prevent you from memory-leaks.
```

Comment:
> With the use of `promises` and `async/await` we are able to catch **any** synchronous error and respond it back to the callee doesn't matter whether it's correct behavior or a syntax error. This is an anti-pattern because any unpredictable application state should exit your server so that it can come back in a clear state. This is useful for debugging and can prevent you from memory-leaks.

=>

> With the use of `promises` and `async/await` we are able to catch **any** synchronous error and return it back to the callee. It doesn't matter whether it's correct behavior or a syntax error. This is an anti-pattern because any unpredictable application state should exit your server so that it can come back in a clear state. This is useful for debugging and can prevent memory-leaks.

I'm also not clear on this justification.